lib/nfsbase.pm

```diff
@@ -0,0 +1,146 @@
+    }
+}
+
+sub try_nfsv2 {
+    # Try that NFSv2 is disabled by default
+    systemctl 'restart nfsserver';
+    assert_script_run "cat /proc/fs/nfsd/versions | grep '\\-2'";
+    assert_script_run "sed -i 's/MOUNTD_OPTIONS=.*/MOUNTD_OPTIONS=\"-V2\"/g' /etc/sysconfig/nfs";
+    assert_script_run "sed -i 's/NFSD_OPTIONS=.*/NFSD_OPTIONS=\"-V2\"/g' /etc/sysconfig/nfs";
+    systemctl 'restart nfsserver';
+    assert_script_run "cat /proc/fs/nfsd/versions | grep '+2'";
+
+    # Disable NFSv2 again
+    assert_script_run "sed -i 's/MOUNTD_OPTIONS=.*/MOUNTD_OPTIONS=\"\"/g' /etc/sysconfig/nfs";
+    assert_script_run "sed -i 's/NFSD_OPTIONS=.*/NFSD_OPTIONS=\"\"/g' /etc/sysconfig/nfs";
+    systemctl 'stop nfsserver';
+}
+
+sub prepare_exports {
+    # Create a directory and place a test file in it
+    assert_script_run 'mkdir /srv/nfs && echo success > /srv/nfs/file.txt';
+
+    # Create also hardlink, symlink and do bindmount
+    assert_script_run '( mkdir /srv/dir /srv/nfs/bindmounteddir && ln -s /srv/dir /srv/nfs/symlinkeddir && mount --bind /srv/dir /srv/nfs/bindmounteddir )';
+    assert_script_run '( echo example > /srv/dir/example && ln /srv/dir/example /srv/nfs/hardlinkedfile & ln -s /srv/dir/example /srv/nfs/symlinkedfile )';
+    assert_script_run 'echo secret > /srv/nfs/secret.txt && chmod 740 /srv/nfs/secret.txt';
+
+    # Create large file and count its md5sum
+    assert_script_run 'fallocate -l 1G /srv/nfs/random';
+    assert_script_run "md5sum /srv/nfs/random | cut -d' ' -f1 > /srv/nfs/random.md5sum";
+}
+
+sub yast_handle_firewall {
+    if (is_sle('<15')) {
+        send_key 'alt-f';    # Open port in firewall
+        assert_screen 'nfs-firewall-open';
+    }
+    else {
+        sleep 1;
+        save_screenshot;
+    }
+}
+
+sub add_rw_share {
+    # Add rw share
+    send_key 'alt-d';
+    assert_screen 'nfs-new-share';
+    type_string '/srv/nfs';
+    send_key 'alt-o';
+
+    # Permissions dialog
+    assert_screen 'nfs-share-host';
+    send_key 'tab';
+    # Change 'ro,root_squash' to 'rw,fsid=0,no_root_squash,...'
+    send_key 'home';
+    send_key 'delete';
+    send_key 'delete';
+    send_key 'delete';
+    type_string "rw,fsid=0,no_";
+    send_key 'alt-o';
+
+    # Saved
+    sleep 1;
+    assert_screen 'nfs-share-saved';
+}
+
+sub mount_export {
+    script_run 'mount|grep nfs';
+    assert_script_run 'cat /etc/fstab | grep nfs';
+
+    # script_run is using bash return logic not perl logic, 0 is true
+    if ((script_run('grep "success" /tmp/nfs/client/file.txt', 90)) != 0) {
+        record_soft_failure 'boo#1006815 nfs mount is not mounted';
+        assert_script_run 'mount /tmp/nfs/client';
+        assert_script_run 'grep "success" /tmp/nfs/client/file.txt';
+    }
+}
+
+sub client_common_tests {
+    # remove added nfs from /etc/fstab
+    assert_script_run 'sed -i \'/nfs/d\' /etc/fstab';
+
+    # compare saved and current fstab, should be same
+    if ((script_run 'diff -b /etc/fstab fstab_before') != 0) {
```

Comment:
what if something else will be changed in a file ? maybe we need some extra check here to decide that actually comments are missing ?

Reply:
I just removed this if completely and used `assert_screen_run` instead because of this bug has been closed several months ago.

Reply:
https://bugzilla.suse.com/show_bug.cgi?id=429326#c44